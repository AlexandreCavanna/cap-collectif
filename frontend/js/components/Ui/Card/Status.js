// @flow
import styled, { type StyledComponent } from 'styled-components';
import * as React from 'react';
import { colorsData } from '../../../utils/colors';
import type { ProposalStepStatusColor } from '~relay/ProposalCollectStatus_proposal.graphql';

type Props = {
  bgColor: ProposalStepStatusColor,
  children: React.Node,
};

const Container: StyledComponent<Props, {}, HTMLDivElement> = styled.div`
  border-bottom-right-radius: 3px;
  border-bottom-left-radius: 3px;
  padding: 3px;
  min-height: 25px;
  color: white;
  font-size: 14px;
  text-align: center;
  background-color: ${props => props.bgColor};
`;

export const Status = (props: Props) => {
  const { bgColor, children } = props;

  const getBgColor = colorsData[bgColor];

  return (
    <Container
      bsStyle={bgColor}
      bgColor={getBgColor}
      className={
        props.bgColor === 'PRIMARY'
          ? 'ellipsis card__status custom-primary-bgcolor'
          : 'ellipsis card__status'
      }>
      {children}
    </Container>
  );
};

Status.defaultProps = {
  bgColor: 'DEFAULT',
};

export default Status;
