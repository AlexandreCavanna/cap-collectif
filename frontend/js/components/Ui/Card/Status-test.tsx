/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Status } from './Status'

describe('<Status />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Status>My Status</Status>)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly with other bgColor', () => {
    const wrapper = shallow(<Status bgColor="DANGER">My Status</Status>)
    expect(wrapper).toMatchSnapshot()
  })
})
