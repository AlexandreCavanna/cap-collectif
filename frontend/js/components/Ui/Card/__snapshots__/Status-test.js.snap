// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should render correctly 1`] = `
<Status__Container
  bgColor="#707070"
  bsStyle="DEFAULT"
  className="ellipsis card__status"
>
  My Status
</Status__Container>
`;

exports[`<Status /> should render correctly with other bgColor 1`] = `
<Status__Container
  bgColor="#f75d56"
  bsStyle="DANGER"
  className="ellipsis card__status"
>
  My Status
</Status__Container>
`;
